### PR TITLE
use safe strings in upnp

### PR DESCRIPTION
**Pull request by zakird**

### CODE REVIEW

File: src/fieldset.c [hunk 1]
```diff
@@ -206,6 +206,16 @@ void fs_chkadd_string(fieldset_t *fs, const char *name, char *value, int free_)
 	}
 }
 
+void fs_chkadd_unsafe_string(fieldset_t *fs, const char *name, char *value, int free_)
+{
+	if (value) {
+		fs_add_unsafe_string(fs, name, value, free_);
```
Comment (dadrian) [inline src/fieldset.c]: This will currently add empty strings. Not sure if that's intended behavior or not.

Comment (zakird) [inline src/fieldset.c]: Empty strings are empty strings and should be treated like empty strings.

**Comment by dadrian** [comment]
Looks fine, aside from the one comment about empty strings.

File: src/fieldset.c (hunk 1 above)
Comment (dadrian) [inline src/fieldset.c]: To clarify: I mean empty strings do not result in an `fs_add_null`, which I'm not sure if is intended behavior or not.

Comment (zakird) [inline src/fieldset.c]: right, which is what's semantically correct.

Comment (dadrian) [inline src/fieldset.c]: Great. :shipit:

**Comment by dadrian** [comment]
Assuming it builds, let's merge.

**Comment by paul-pearce** [comment]
This is for issue https://github.com/zmap/zmap/issues/329

**Pull request merged**